calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -0,0 +1,265 @@
+        for case, is_valid in cases:
+            print "Testing case %s ..." % case
+            bgp_validate_arguments(case)
+            if is_valid:
+                assert not m_sys_exit.called
+            else:
+                assert m_sys_exit.called
+            m_sys_exit.reset_mock()
+
+    @patch('calico_ctl.bgp.check_ip_version',autospec=True)
+    @patch('calico_ctl.bgp.BGPPeer',autospec=True)
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_bgp_peer_add(self,m_client,m_BGPPeer,m_check_ipv):
+        """
+        Test bgp_peer_add function for calico_ctl bgp
+        """
+        # Set up mock objects
+        address = '1.2.3.4'
+        peer =  Mock(spec=BGPPeer)
+        m_check_ipv.return_value = address
+        m_BGPPeer.return_value = peer
+
+        # Call method under test
+        bgp_peer_add('1.2.3.4','v4',1)
+
+        # Assert
+        m_check_ipv.assert_called_once_with('1.2.3.4','v4',IPAddress)
+        m_BGPPeer.assert_called_once_with(address,1)
+        m_client.add_bgp_peer.assert_called_once_with('v4',peer)
+
+    @patch('calico_ctl.bgp.check_ip_version',autospec=True)
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_bgp_peer_remove(self,m_client,m_check_ipv):
+        """
+        Test bgp_peer_remove function for calicoctl bgp
+        """
+        # Set up mock objects
+        address = '1.2.3.4'
+        m_check_ipv.return_value = address
+
+        # Call method under test
+        bgp_peer_remove('1.2.3.0','v4')
+
+        # Assert
+        m_check_ipv.assert_called_once_with('1.2.3.0','v4',IPAddress)
+        m_client.remove_bgp_peer.assert_called_once_with('v4',address)
+
+    @patch('calico_ctl.bgp.check_ip_version',autospec=True)
+    @patch('calico_ctl.bgp.client',autospec=True)
+    @patch('calico_ctl.bgp.sys.exit',autospec=True)
+    def test_bgp_peer_remove_fail(self, m_sys_exit, m_client, m_check_ipv):
+        """
+        Test for failure of bgp_peer_remove function
+            for calicoctl bgp
+        """
+        # Set up mock objects
+        address = '1.2.3.4'
+        m_check_ipv.return_value = address
+        m_client.remove_bgp_peer.side_effect = KeyError
+
+        # Call method under test
+        bgp_peer_remove('1.2.3.0','v4')
+
+        # Assert
+        m_check_ipv.assert_called_once_with('1.2.3.0','v4',IPAddress)
+        m_client.remove_bgp_peer.assert_called_once_with('v4',address)
+        m_sys_exit.assert_called_once_with(1)
+
+    def test_bgp_peer_show(self):
+        pass
+
+    @patch('calico_ctl.bgp.client',autospec=True)
+    def test_set_default_node_as(self,m_client):
+        """
+        Test set_default_node_as function for calicoctl bgp
+        """
+        # Call method under test
+        set_default_node_as(1)
+
+        # Assert
+        m_client.set_default_node_as.\
+            assert_called_once_with(1)
+
+    @patch('calico_ctl.bgp.client',autospec=True)
+    @patch('calico_ctl.bgp.sys.stdout',new_callable=StringIO)
```

Comment:
This can be inported just as sys.stdout.

The times where we would want to import a specific class/function are if the class/function was imported in the file, and even then only if it was imported like this:
`from sys import stdout`
Since is bgp.py just imports sys, it does not take ownership of any of the sys fields or values, so this patch could just be `sys.stdout`.